File: src/AirWaterStore/WebApps/AirWaterStore.Web/Models/Chat/Message.cs
namespace AirWaterStore.Web.Models.Chat;

public partial class Message
{
    public int MessageId { get; set; }

    public string ChatRoomId { get; set; } = "";

    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public DateTime? SentAt { get; set; }
}

public record GetMessagesByChatRoomIdResponse(IEnumerable<Message> Messages);

public record CreateMessageRequest(
    string ChatRoomId,
    int UserId,
    string Content,
    DateTime SentAt
    );

public record CreateMessageResponse(string Id);

File: src/AirWaterStore/Services/Chat/Chat.API/app/schema/message.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict, AliasGenerator
from pydantic.alias_generators import to_camel


class MessageResponse(BaseModel):
    message_id: str
    chat_room_id: str
    user_id: int
    content: str
    sent_at: datetime

    model_config = ConfigDict(
        alias_generator=AliasGenerator(serialization_alias=to_camel),
    )


class MessageCreateRequest(BaseModel):
    user_id: int
    chat_room_id: str
    content: str

    model_config = ConfigDict(
        alias_generator=AliasGenerator(validation_alias=to_camel),
        populate_by_name=True,
    )


class MessageResponseEnvelope(BaseModel):
    message: MessageResponse
    model_config = ConfigDict(
        alias_generator=AliasGenerator(serialization_alias=to_camel),
    )


class MessageListResponseEnvelope(BaseModel):
    messages: List[MessageResponse]

    model_config = ConfigDict(
        alias_generator=AliasGenerator(serialization_alias=to_camel),
    )
